src/qpmodel.jl

```diff
@@ -54,76 +66,108 @@ end
+function obj(qp :: AbstractQuadraticModel, x :: AbstractVector)
     v = qp.data.opH * x
     f = qp.data.c0 + dot(qp.data.c, x) + 0.5 * dot(v, x)
     qp.counters.neval_hprod += 1
     f
 end
 
-function grad(qp::AbstractQuadraticModel, x::AbstractVector)
+function grad(qp :: AbstractQuadraticModel, x :: AbstractVector)
     g = Vector{eltype(x)}(undef, qp.meta.nvar)
     grad!(qp, x, g)
 end
 
-function grad!(qp::AbstractQuadraticModel, x::AbstractVector, g::AbstractVector)
+function grad!(qp :: AbstractQuadraticModel, x :: AbstractVector, g :: AbstractVector)
     v = qp.data.opH * x
     @. g = qp.data.c + v
     qp.counters.neval_hprod += 1
     g
 end
 
-hess_coord(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = findnz(qp.data.H)
+hess_coord(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = findnz(qp.data.H)
 
-hess(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.H
+hess(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.H
 
-hess_op(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.opH
+hess_op(qp :: AbstractQuadraticModel, ::AbstractVector; kwargs...) = qp.data.opH
 
-function cons(qp::AbstractQuadraticModel, x::AbstractVector)
+function cons(qp::AbstractQuadraticModel, x :: AbstractVector)
     c = Vector{eltype(x)}(undef, qp.meta.ncon)
     cons!(qp, x, c)
 end
 
-function cons!(qp::AbstractQuadraticModel, x::AbstractVector, c::AbstractVector)
+function cons!(qp :: AbstractQuadraticModel, x :: AbstractVector, c :: AbstractVector)
     mul!(c, qp.data.A, x)
     qp.counters.neval_jprod += 1
     c
 end
 
-jac_coord(qp::AbstractQuadraticModel, ::AbstractVector; kwargs...) = findnz(qp.data.A)
+"""
+Return the structure of the constraint's Jacobian in sparse coordinate format in place.
+"""
+function jac_structure!(qp :: QuadraticModel, rows :: Vector{<: Integer}, cols :: Vector{<: Integer}; kwargs...)
+    rows .= qp.data.A.rowval
+    cols .= findnz(qp.data.A)[2]
+end
+
+"""
+Return the structure of the Lagrangian Hessian in sparse coordinate format in place.
+"""
+function hess_structure!(qp :: QuadraticModel, rows :: Vector{<: Integer}, cols :: Vector{<: Integer}; kwargs...)
+    rows .= qp.data.H.rowval
+    cols .= findnz(qp.data.H)[2]
+end
+
+"""
+Return the structure of the constraint's Jacobian in sparse coordinate format in place.
+"""
+function jac_coord!(qp :: QuadraticModel, x :: AbstractVector, rows :: Vector{<: Integer},
+                    cols :: Vector{<: Integer}, vals :: Vector{<: AbstractFloat}; kwargs...)
+                    vals .= findnz(qp.data.A)[3]
+end
+
+"""
+Evaluate the Lagrangian Hessian at `x` in sparse coordinate format. Only the lower triangle is returned.
+"""
+function hess_coord!(qp :: QuadraticModel, :: AbstractVector, rows :: AbstractVector{<: Integer},
+                     cols :: AbstractVector{<: Integer}, vals :: Vector{<: AbstractFloat}; kwargs...)
+                     vals .= findnz(qp.data.H)[3]
```

Comment:
Indentation is wrong.

Reply:
And the result of `vals` depends on the keyword argument `obj_weight`.

Reply:
Which makes me notice that `hess` and `hess_op` also depend on it.